protocol/test-sequencing.md

```diff
@@ -0,0 +1,216 @@
+# Summary
+
+Interop testing requires completion of a long
+[checklist](https://github.com/ethereum-optimism/optimism/issues/13855) of particularly constructed blocks and reorgs.
+To implement these tests, we need new tooling that can include transactions and perform reorgs in exact ways.
+
+While implementing the test tooling, we have an opportunity to prepare separation of sequencing from the op-node,
+to reduce op-node complexity and encapsulate sequencing concerns better.
+
+# Problem Statement + Context
+
+Sequencing in the op-stack can be bucketed into different types:
+- Legacy sequencing (tx-pool, op-node with op-geth, recommit, priority fee)
+- Flashblocks (rollup-boost proxy between op-node and op-geth, mux to builder node, 250ms incremental block building)
+- Testing (action-tests, next-up interop work)
+
+The testing type of work is in need of change, to provide the level of block-building control
+that [action-tests](https://github.com/ethereum-optimism/optimism/tree/develop/op-e2e/actions)
+have to fully end-to-end tests like those backed by
+[Kurtosis](https://github.com/ethereum-optimism/optimism/tree/develop/kurtosis-devnet).
+
+Instead of building a limited testing tool, we can consider building a full sequencer service:
+- Op-node complexity can be reduced.
```

Comment:
what is the difference between this tool and what `rbuilder` does?

Reply:
- ability to reorg, important for a lot of the reorg tests.
- ability to bypass validation rules when/if desired, e.g. for the invalid-message test cases.
- ability to build blocks when desired, rather than on the start/stop command of a real sequencer.
- ability to synchronously confirm txs with RPC calls, rather than submitting them to a tx-pool.